text/0000-glimmer-components.md

```diff
@@ -0,0 +1,1019 @@
+
+```hbs
+<!-- templates/components/post.hbs -->
+<section ...attributes role="region" type={{@post.type}} class="post {{@post.type}}">
+  {{#if (eq @post.type 'image')}}
+    <img {{did-render this.didInsertImage}} src={{@post.imageUrl}} title={{@post.imageTitle}} />
+  {{/if}}
+
+  {{@post.text}}
+</section>
+```
+```js
+// components/post.js
+export default class PostComponent extends GlimmerComponent {
+  didInsertImage(element) {
+    setupImageOverlay(element);
+  }
+}
+```
+
+Glimmer components eliminate many of the common paper cuts that cause confusion
+with classic components, and align more closely with modern template syntax and
+features.
+
+### Outer HTML Semantics
+
+The biggest change Glimmer components make is defaulting to outer HTML
+semantics. In the classic component API, components had a implicit wrapper
+element. Given this component template:
+
+```hbs
+Hello, world!
+```
+
+The output by _default_ would be something like:
+
+```html
+<!-- OUTPUT -->
+<div id="ember-1234" class="ember-view">
+  Hello, world!
+</div>
+```
+
+But we can't know that for sure unless we look at the component definition. If
+we do, we might see that the outer wrapping element is actually a `section`, and
+it has a `.hello-world` class:
+
+```js
+export default Component.extend({
+  tagName: 'section',
+  classNames: ['hello-world']
+});
+```
+```html
+<!-- OUTPUT -->
+<section id="ember-1234" class="hello-world ember-view">
+  Hello, world!
+</section>
+```
+
+This behavior means that the template for a component is missing crucial
+information and context. Even for the simplest component, users must check the
+class definition to know with certainty what the full template of the component
+is. And unlike bindings, there is no hint to the user that there may be
+something dynamic that they should check on - without advanced knowledge of
+Ember's APIs, there is no way of knowing about this behavior.
+
+By contrast, Glimmer components have no wrapping outer element - What you see in
+the template is what you get in the output. There is no way to define class
+names, class name bindings, attribute bindings, or any other DOM element values
+_from the component class_. The template is the single source of truth for the
+output of a component, and any dynamic values are explicitly stated in it.
+
+```hbs
+<!-- template.hbs -->
+<section class="hello-world">
+  Hello, world!
+</section>
+```
+
+```html
+<!-- OUTPUT -->
+<section class="hello-world">
+  Hello, world!
+</section>
+```
+
+We can immediately see that this is a static component with no bindings, no
+dynamic values, and no meaningful state. Even if there was a component
+definition, we know that it is not in any way affecting the output of this
+template. Special element ids and classes are also not present, making the
+output appear less magical.
+
+This small change makes a huge difference in the large: users can spend less
```

Comment:
I made [this connection](https://en.wikipedia.org/wiki/Programming_in_the_large_and_programming_in_the_small#Programming_in_the_large) for "in the large"
```suggestion
This micro change makes a macro difference: users can spend less
```